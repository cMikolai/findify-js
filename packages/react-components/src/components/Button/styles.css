.button {
  border: none;
  outline: none;
  background: transparent;
  cursor: pointer;
  transition: background-color .1s ease-in;
  padding: 0;
  margin: 0;
}

.raw{
  display: block;
  padding: 0;
  margin: 0;
  display: block;
  text-align: left;
  background: transparent;
  height: auto;
}
