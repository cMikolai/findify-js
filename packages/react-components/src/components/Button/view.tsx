import React from 'react';
import { compose, withHandlers } from 'recompose';
import cx from 'classnames';

export default ({ theme, onClick, children, active, size, className }) => (
  <button
    onClick={onClick}
    className={cx(
      theme.button,
      active && theme.active,
      theme[size],
      className
    )}
  >
  { children }
  </button>
);
