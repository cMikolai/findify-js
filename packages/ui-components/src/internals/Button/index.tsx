import * as React from 'react';
import * as cx from 'classnames';
const styles = require('./styles.css');

export interface Props {
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  children?: React.ReactChild;
  className?: string;
  isLoading: boolean;
}

const getClassName = (props: Props) =>
  cx(styles.root, props.className, {
    [styles.loading]: props.isLoading,
  });

export const Button = (props: Props) => (
  <button onClick={props.onClick} className={getClassName(props)}>
    {props.children}
  </button>
);

export default Button;
