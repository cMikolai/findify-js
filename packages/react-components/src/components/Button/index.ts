import { compose, setDisplayName } from 'recompose';
import withTheme from 'helpers/withTheme';

import view from './view';
import styles from './styles.css';

export default compose(
  setDisplayName('Button'),
  withTheme(styles)
)(view);
