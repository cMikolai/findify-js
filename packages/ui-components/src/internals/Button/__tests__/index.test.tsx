import * as React from 'react';
import { shallow, render } from 'enzyme';
import Button, { Props } from '..';

describe('<Button />', () => {
  it('renders correctly', () => {
    const properties: Props[] = [
      {},
      { className: 'foo' },
      {
        className: 'bar',
        children: <div>children div</div>,
      },
    ];
    properties.forEach((props: Props) => {
      const component = render(<Button {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
