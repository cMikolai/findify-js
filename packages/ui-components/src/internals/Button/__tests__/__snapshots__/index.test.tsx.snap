// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> attaches and calls onClick event handler 1`] = `
<button
  className="root"
  onClick={[Function]}
/>
`;

exports[`<Button /> renders correctly 1`] = `
<button
  class="root"
/>
`;

exports[`<Button /> renders correctly 2`] = `
<button
  class="root foo"
/>
`;

exports[`<Button /> renders correctly 3`] = `
<button
  class="root bar"
>
  <div>
    children div
  </div>
</button>
`;
